javatests/arcs/core/allocator/AllocatorTestBase.kt

```diff
@@ -538,4 +555,90 @@ open class AllocatorTestBase {
             ArcState.Error
         )
     }
+
+    @Test
+    fun allocator_logArcStartAndStopToAnalytics() = runAllocatorTest {
+        val arcId = allocator.startArcForPlan(
+            "readWriteParticle",
+            PersonPlan
+        )
+
+        // Verify logs are recorded.
+        verify(mockLogger, times(1)).logAllocatorEvent(
+            eq(Analytics.Event.StartArc), eq("readWriteParticle"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StartArc), contains("arcs.core.host.TestingJvmProdHost"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StartArc), contains("arcs.core.allocator.AllocatorTestBase.ReadingHost"), any())
```

Comment:
These checks won't work, unless you put this test on CoreAllocatorTest only.  Putting tests on AllocatorTestBase tests them on all platforms. On AndroidAllocatorTests, the ArcHosts come from TestReadingExternalHostService for example. 

If you want to do this in a platform independent manner, use readingExternalHost::class to obtain the class names for example. readingExternalHost/writingExternalHost/pureHost contain the instances of the ArcHosts that are based on platform overrides.